### PR TITLE
Use more ES6 in node code.

### PR DESCRIPTION
**Pull request by jdalton**
This PR uses more ES6 in node code.

### CODE REVIEW

File: repl/index.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-    return;
-  }
-
-  if (method.reified) {
+  if (typeof method !== "function" ||
+      method.reified) {
     return;
   }
 
   vm[name] = function (code) {
```
Comment (jdalton) [inline repl/index.js]: Interesting that if I make, `vm[name] = (code) => {`, an arrow function tests fail.
Do you know if it's inspecting the toString of these functions or if they require them to have a `.prototype`?

Comment (benjamn) [inline repl/index.js]: I'll investigate.

File: node/caching-compiler.js [hunk 2]
```diff
@@ -162,17 +166,16 @@ function readPkgInfo(dir) {
...
       // reification even if "reify" is listed as a dependency.
       return null;
     }
 
-    function check(name) {
-      return typeof pkg[name] === "object" &&
-        hasOwn.call(pkg[name], "reify");
-    }
+    const check = (name) => (
+      typeof pkg[name] === "object" && hasOwn.call(pkg[name], "reify")
```
Comment (benjamn) [inline node/caching-compiler.js]: Note to self: add a `;` here.

Comment (jdalton) [inline node/caching-compiler.js]: Yes! On my todo is to add some linting.

File: repl/index.js (hunk 1 above)
Comment (jdalton) [inline repl/index.js]: Ya, it's not critical to this PR just curious (seems interesting).

File: node/caching-compiler.js (hunk 2 above)
Comment (benjamn) [inline node/caching-compiler.js]: Oh wait, that's an expression-bodied arrow function. Never mind!

File: repl/index.js (hunk 1 above)
Comment (jdalton) [inline repl/index.js]: Ah, it's because arrow functions don't have `arguments` objects so it's referencing the `arguments` object of the module itself.

**Pull request merged**